refactor: improve readability

sazk07 opened the pull request:
Fix typo in condition check

<!--
Thank you for your pull request. Please provide a description and 
note the Certificate of Origin below. 

-->

<!--
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

sazk07 committed:
Fix typo: Update index.js

Fix typo in condition check

wesleytodd commented on examples/params/index.js:
```diff
@@ -32,7 +32,7 @@ app.param(['to', 'from'], function(req, res, next, num, name){
 
 app.param('user', function(req, res, next, id){
-  if (req.user = users[id]) {
+  if (req.user === users[id]) {
```
While this is some bad code, I don't think it is a typo.

This check is checking if users array has an id from the param value, and then evaluating the truthiness of `req.user`. For sure something we should remove as unclear, but not a typeo in the sense that this change fixes it.

krzysdz replied:
Also is the only place that sets `req.user`, which is then read in line 55. Changing the assignment to comparison will break this example.

sazk07 committed:
Fix: separate assignment and condition check in Update index.js

assignment was being done inside condition check leading to confusion in reading code.

sazk07 commented:
assignment inside condition check leading to confusion in reading code. Change proposed.

sazk07 reviewed (commented):
updated

UlisesGascon commented:
@sazk07 do you want to backport this to [v4 branch](https://github.com/expressjs/express/tree/4.x) too?

sazk07 commented:
> @sazk07 do you want to backport this to [v4 branch](https://github.com/expressjs/express/tree/4.x) too?

@UlisesGascon PR made in v4 branch. pls check from your end.
